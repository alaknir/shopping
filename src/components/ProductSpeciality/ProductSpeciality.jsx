import React from "react";

const ProductSpeciality = ({ isExclusive, isSale }) => (
  <div className="row product-spl">
    <div className="col-xs-12 product-spl__badge">
      {isSale && <span className="product-spl__badge-sale">Sale</span>}
      {isExclusive && (
        <span className="product-spl__badge-exclusive">Exclusive</span>
      )}
    </div>
  </div>
);

ProductSpeciality.defaultProps = {
  isExclusive: false,
  isSale: false
};

export default ProductSpeciality;
