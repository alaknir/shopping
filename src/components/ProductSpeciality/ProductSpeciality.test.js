import React from "react";
import { mount } from "enzyme";

import ProductSpeciality from "./ProductSpeciality";

describe("ProductSpeciality ", () => {
  it("Should render properly isExclusive", () => {
    const component = <ProductSpeciality isExclusive />;
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render properly isSale", () => {
    const component = <ProductSpeciality isSale />;
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });
});
