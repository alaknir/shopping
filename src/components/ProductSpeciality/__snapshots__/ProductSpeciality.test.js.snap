// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSpeciality  Should render properly isExclusive 1`] = `
<ProductSpeciality
  isExclusive={true}
  isSale={false}
>
  <div
    className="row product-spl"
  >
    <div
      className="col-xs-12 product-spl__badge"
    >
      <span
        className="product-spl__badge-exclusive"
      >
        Exclusive
      </span>
    </div>
  </div>
</ProductSpeciality>
`;

exports[`ProductSpeciality  Should render properly isSale 1`] = `
<ProductSpeciality
  isExclusive={false}
  isSale={true}
>
  <div
    className="row product-spl"
  >
    <div
      className="col-xs-12 product-spl__badge"
    >
      <span
        className="product-spl__badge-sale"
      >
        Sale
      </span>
    </div>
  </div>
</ProductSpeciality>
`;
